lib/uring/uring_stubs.c

```diff
@@ -405,6 +405,46 @@ ocaml_uring_extract_sockaddr(value v) {
   CAMLreturn(v_sockaddr);
 }
 
+#define Msghdr_val(v) (*((struct msghdr **) Data_custom_val(v)))
+
+static void finalize_msghdr(value v) {
+  caml_stat_free(Msghdr_val(v));
+  Msghdr_val(v) = NULL;
+}
+
+static struct custom_operations msghdr_ops = {
+  "uring.msghdr_ops",
+  finalize_msghdr,
+  custom_compare_default,
+  custom_hash_default,
+  custom_serialize_default,
+  custom_deserialize_default,
+  custom_compare_ext_default,
+  custom_fixed_length_default
+};
+
+// allocate a custom block containing a sendmsg/recvmsg msghdr 
```

Comment:
This should note that the arguments cannot be freed before the result it, as it contains pointers to them.